### PR TITLE
Auth: Decouple client and hook registration

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
Currently all client and hook registrations happens in the service provider factory. This prevent us from injecting this authn into other services that needs it due to circular injections. We decided early on to move the registrations into their respected service but never got around to it.

As a first step I created a new component that depends on `authn.Service` and all other services. The only think this component does is register clients and hooks and then return an empty struct. I then use the backgroudservice injection hack to make sure this is run on startup.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/authn/authnimpl/registration.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		if !cfg.DisableLoginForm {
+			authnSvc.RegisterClient(clients.ProvideForm(passwordClient))
+		}
+	}
+
+	if cfg.AuthProxy.Enabled && len(proxyClients) > 0 {
+		proxy, err := clients.ProvideProxy(cfg, cache, proxyClients...)
+		if err != nil {
+			// TODO: Fix logging
+			// s.log.Error("Failed to configure auth proxy", "err", err)
```
Comment (IevaVasiljeva) [inline pkg/services/authn/authnimpl/registration.go:75]: This TODO probably should be resolved before merging.

**Review by IevaVasiljeva — APPROVED** [review]
LGTM with a small comment about fixing logging.

**Review by mgyongyosi — APPROVED** [review]

**Review by Jguer — APPROVED** [review]
if we can hold just a bit to merge OBO, it would save me a lengthy rebase

**Comment by kalleep** [comment]
> if we can hold just a bit to merge OBO, it would save me a lengthy rebase

That is why I have not merged this one yet. I already have caused some conflicts for you :see_no_evil:

**Comment by alexanderzobnin** [comment]
We still have problems here I think. Authn service depends on `usagestats.Service` and `login.AuthInfoService` and they are depend on accesscontrol service. Which makes it impossible to inject authn service into access control one.

**Comment by kalleep** [comment]
@alexanderzobnin I managed to get rid of usage stats as a dependency but the other problem is AuthInfoService depends on SecretService that depends on UsageStats. Maybe the proper fix if to remove accesscontrol.Service as a dependecy in usage stats. The only think it does is register a fixed role and we can probably just move that declaration and registration into access control package instead, WDYT?

**Review by alexanderzobnin — APPROVED** [review]

**Pull request merged**